go/store/datas/pull/pull_chunk_tracker_test.go

```diff
@@ -0,0 +1,169 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pull
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/dolthub/dolt/go/store/hash"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPullChunkTracker(t *testing.T) {
+	t.Run("Empty", func(t *testing.T) {
+		tracker := NewPullChunkTracker(context.Background(), make(hash.HashSet), TrackerConfig{
+			BatchSize:          64 * 1024,
+			HasManyThreadCount: 3,
+			Haser:              nil,
+		})
+		hs, ok, err := tracker.GetChunksToFetch()
+		assert.Len(t, hs, 0)
+		assert.False(t, ok)
+		assert.NoError(t, err)
+		tracker.Close()
+	})
+
+	t.Run("HasAllInitial", func(t *testing.T) {
+		hs := make(hash.HashSet)
+		for i := byte(0); i < byte(10); i++ {
+			var h hash.Hash
+			h[0] = i
+			hs.Insert(h)
+		}
+		tracker := NewPullChunkTracker(context.Background(), hs, TrackerConfig{
+			BatchSize:          64 * 1024,
+			HasManyThreadCount: 3,
+			Haser:              hasAllHaser{},
+		})
+		hs, ok, err := tracker.GetChunksToFetch()
+		assert.Len(t, hs, 0)
+		assert.False(t, ok)
+		assert.NoError(t, err)
+		tracker.Close()
+	})
+
+	t.Run("HasNoneInitial", func(t *testing.T) {
+		hs := make(hash.HashSet)
+		for i := byte(1); i <= byte(10); i++ {
+			var h hash.Hash
+			h[0] = i
+			hs.Insert(h)
+		}
+		tracker := NewPullChunkTracker(context.Background(), hs, TrackerConfig{
+			BatchSize:          64 * 1024,
+			HasManyThreadCount: 3,
+			Haser:              hasNoneHaser{},
+		})
+		hs, ok, err := tracker.GetChunksToFetch()
+		assert.Len(t, hs, 10)
+		assert.True(t, ok)
+		assert.NoError(t, err)
+		hs, ok, err = tracker.GetChunksToFetch()
+		assert.Len(t, hs, 0)
+		assert.False(t, ok)
+		assert.NoError(t, err)
+
+		for i := byte(1); i <= byte(10); i++ {
+			var h hash.Hash
+			h[1] = i
+			tracker.Seen(h)
+		}
+
+		cnt := 0
+		for {
+			hs, ok, err := tracker.GetChunksToFetch()
+			assert.NoError(t, err)
+			if !ok {
+				assert.Equal(t, 10, cnt)
+				break
+			}
+			cnt += len(hs)
+		}
+
+		tracker.Close()
+	})
+
+	t.Run("HasManyError", func(t *testing.T) {
+		hs := make(hash.HashSet)
+		for i := byte(0); i < byte(10); i++ {
+			var h hash.Hash
+			h[0] = i
+			hs.Insert(h)
+		}
+		tracker := NewPullChunkTracker(context.Background(), hs, TrackerConfig{
+			BatchSize:          64 * 1024,
+			HasManyThreadCount: 3,
+			Haser:              errHaser{},
+		})
+		_, _, err := tracker.GetChunksToFetch()
+		assert.Error(t, err)
+		tracker.Close()
+	})
+
+	t.Run("InitialAreSeen", func(t *testing.T) {
+		hs := make(hash.HashSet)
+		for i := byte(0); i < byte(10); i++ {
+			var h hash.Hash
+			h[0] = i
+			hs.Insert(h)
+		}
+		tracker := NewPullChunkTracker(context.Background(), hs, TrackerConfig{
+			BatchSize:          64 * 1024,
+			HasManyThreadCount: 3,
```

Comment:
I was sure you made this configurable so that you could test is. Not the case? Just make it a constant then?

Reply:
Made it a constant.